proving/proving.go

```diff
@@ -15,63 +14,65 @@ const (
 	LabelGroupSize                          = shared.LabelGroupSize
 	MaxSpace                                = shared.MaxSpace
 	MaxNumOfFiles                           = shared.MaxNumOfFiles
-	MinDifficulty                           = shared.MinDifficulty
-	MaxDifficulty                           = shared.MaxDifficulty
 	LowestLayerToCacheDuringProofGeneration = shared.LowestLayerToCacheDuringProofGeneration
 )
 
-func GenerateProof(id []byte, challenge Challenge, numOfProvenLabels uint8, difficulty Difficulty) (proof Proof,
+type (
+	Difficulty = shared.Difficulty
+	Challenge  = shared.Challenge
+)
+
+func GenerateProof(id []byte, challenge Challenge, numOfProvenLabels uint8, difficulty Difficulty, dir string) (proof *Proof,
 	err error) {
 
-	proof, err = generateProof(id, challenge, numOfProvenLabels, difficulty)
+	proof, err = generateProof(id, challenge, numOfProvenLabels, difficulty, dir)
 	if err != nil {
 		err = fmt.Errorf("proof generation failed: %v", err)
 		log.Error(err.Error())
 	}
 	return proof, err
 }
 
-func generateProof(id []byte, challenge Challenge, numOfProvenLabels uint8, difficulty Difficulty) (proof Proof,
-	err error) {
-
-	err = difficulty.Validate()
+func generateProof(id []byte, challenge Challenge, numOfProvenLabels uint8, difficulty Difficulty, dir string) (*Proof, error) {
```

Comment:
`generateProof` should validate that the initialization had completed before generating new proofs.